Reverts the patch described below. It breaks qtdeclarative build
in very mysterious ways, like:

CMake Error at build/lib/cmake/Qt6QmlTools/Qt6QmlToolsConfig.cmake:56 (include):
  include could not find requested file:

    /builddir/qt6-qtdeclarative-6.10.0/build/lib/cmake/Qt6QmlTools/Qt6QmlToolsTargets.cmake
Call Stack (most recent call first):
  /usr/lib64/cmake/Qt6/QtPublicDependencyHelpers.cmake:100 (find_package)
  build/lib/cmake/Qt6Qml/Qt6QmlDependencies.cmake:39 (_qt_internal_find_tool_dependencies)
  build/lib/cmake/Qt6Qml/Qt6QmlConfig.cmake:53 (include)
  /usr/share/cmake-4.1/Modules/CMakeFindDependencyMacro.cmake:78 (find_package)
  /usr/lib64/cmake/Qt6/QtPublicDependencyHelpers.cmake:142 (find_dependency)
  build/lib/cmake/Qt6QmlPrivate/Qt6QmlPrivateDependencies.cmake:45 (_qt_internal_find_qt_dependencies)
  build/lib/cmake/Qt6QmlPrivate/Qt6QmlPrivateConfig.cmake:46 (include)
  /usr/lib64/cmake/Qt6/Qt6Config.cmake:243 (find_package)
  src/qml/doc/snippets/qmltc/CMakeLists.txt:62 (find_package)

CMake Error at build/lib/cmake/Qt6QmlTools/Qt6QmlToolsAdditionalTargetInfo.cmake:9 (set_property):
  set_property can not be used on an ALIAS target.
Call Stack (most recent call first):
  build/lib/cmake/Qt6QmlTools/Qt6QmlToolsConfig.cmake:57 (include)
  /usr/lib64/cmake/Qt6/QtPublicDependencyHelpers.cmake:100 (find_package)
  build/lib/cmake/Qt6Qml/Qt6QmlDependencies.cmake:39 (_qt_internal_find_tool_dependencies)
  build/lib/cmake/Qt6Qml/Qt6QmlConfig.cmake:53 (include)
  /usr/share/cmake-4.1/Modules/CMakeFindDependencyMacro.cmake:78 (find_package)
  /usr/lib64/cmake/Qt6/QtPublicDependencyHelpers.cmake:142 (find_dependency)
  build/lib/cmake/Qt6QmlPrivate/Qt6QmlPrivateDependencies.cmake:45 (_qt_internal_find_qt_dependencies)
  build/lib/cmake/Qt6QmlPrivate/Qt6QmlPrivateConfig.cmake:46 (include)
  /usr/lib64/cmake/Qt6/Qt6Config.cmake:243 (find_package)
  src/qml/doc/snippets/qmltc/CMakeLists.txt:62 (find_package)

From dad91592e609364231decaba771418d2d48d4910 Mon Sep 17 00:00:00 2001
From: Sze Howe Koh <szehowe.koh@gmail.com>
Date: Thu, 14 Aug 2025 16:16:17 +0800
Subject: [PATCH] Doc: Demonstrate finding some private modules

Explicitly finding a private module was not strictly required in Qt
6.9 and older. This patch adds a small demonstration to some snippets,
to show how to find QmlPrivate/QuickPrivate. This might be helpful
for users who're scratching their heads when their previously-working
project now fails to build in Qt 6.10.

Change-Id: Ia8a08d94ea65aa73a44c7a7610d28119582ebf0f
Reviewed-by: Fabian Kosmale <fabian.kosmale@qt.io>
Reviewed-by: Ulf Hermann <ulf.hermann@qt.io>
Reviewed-by: Alexandru Croitor <alexandru.croitor@qt.io>
(cherry picked from commit 5296c0255b83b8a92e82b3ec4d536655742b90cf)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/qml/doc/snippets/qmltc/CMakeLists.txt                       | 1 +
 .../src/tools/qtquickcompiler/qtqml-qml-script-compiler.qdoc    | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/src/qml/doc/snippets/qmltc/CMakeLists.txt b/src/qml/doc/snippets/qmltc/CMakeLists.txt
index 93477a7..4574a44 100644
--- a/src/qml/doc/snippets/qmltc/CMakeLists.txt
+++ b/src/qml/doc/snippets/qmltc/CMakeLists.txt
@@ -59,6 +59,5 @@ qt6_add_qml_module(${application_name}
 
 #! [qmltc-compile-to-cpp]
 # (qmltc-specific) Link *private* libraries that correspond to QML modules:
-find_package(Qt6 COMPONENTS QmlPrivate QuickPrivate)
 target_link_libraries(${application_name} PRIVATE Qt::QmlPrivate Qt::QuickPrivate)
 #! [qmltc-compile-to-cpp]
diff --git a/src/qml/doc/src/tools/qtquickcompiler/qtqml-qml-script-compiler.qdoc b/src/qml/doc/src/tools/qtquickcompiler/qtqml-qml-script-compiler.qdoc
index d31a7a8..174768d 100644
--- a/src/qml/doc/src/tools/qtquickcompiler/qtqml-qml-script-compiler.qdoc
+++ b/src/qml/doc/src/tools/qtquickcompiler/qtqml-qml-script-compiler.qdoc
@@ -88,8 +88,6 @@ To enable direct mode, you should consider the followings:
 
     \li Link all the relavant private Qt modules instead of their public counterparts.
 \badcode
-    find_package(Qt6 COMPONENTS QmlPrivate QuickPrivate)
-
     qt_add_qml_module(someTarget
     ...
     )
